doc/jenkins.md

```diff
@@ -0,0 +1,271 @@
+# Jenkins
+
+Jenkins is used to implement CI/CD pipeline for Mayastor. We make use of jenkins
+pipeline plugin that allows us to maintain CI/CD as a code that is part of
+Mayastor repository (Jenkinsfile). The goal of this document is to describe
+steps needed to set up the jenkins including the test executors (slaves in
+jenkins terminology). Details differ depending which environment the jenkins is
+set up in: AWS EC2 cloud, server with KVM guests, etc. We describe the most
+important common steps for all environments. We make use of NixOS capabilities
+for system configuration of nodes (as opposed to using ansible, salt, etc.).
```

Comment:
Small nit - I think the 'j' in jenkins should be capitalised throughout ("Jenkins")

Reply:
thanks. will do 👍